exp/controllers/rosamachinepool_controller.go

```diff
@@ -178,6 +180,15 @@ func (r *ROSAMachinePoolReconciler) reconcileNormal(ctx context.Context,
 	}
 	defer rosaClient.Close()
 
+	isValid, err := validateMachinePoolSpec(machinePoolScope)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to validate ROSAMachinePool.spec: %w", err)
+	}
+	if !isValid {
+		// dont' requeue because input is invalid and manual intervention is needed.
```

Comment:
We need to expose this in the status, or it's not actionable for users.

Reply:
its exposed in the func
```go
		message := fmt.Sprintf("version %s is not supported, should be in the range: >= %s and <= %s", version, minSupportedVersion, maxSupportedVersion)
		machinePoolScope.RosaMachinePool.Status.FailureMessage = &message
		return false, nil
```
https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/4764/files#diff-ca8ae33b4375b0269fe24617620c4d304d1882cb8d562c7b544331d272c7aec1R352

Reply:
I see - it's surprising that a function named `validate...()` mutates its inputs. Can we make the control flow clearer?